routers/api/v1/api.go

```diff
@@ -295,25 +347,6 @@ func reqToken() func(ctx *context.APIContext) {
 			return
 		}
 
-		if true == ctx.Data["IsApiToken"] {
```

Comment:
I'm not sure whether it is safe to remove it.
How to approve it? Did you check all router paths?

Reply:
Yes. I checked all routers and also add some tests for the newly added code. These code have been extracted to a new method checkTokenPublicOnly.